.ci.yaml

```diff
@@ -14,7 +14,8 @@ platform_properties:
       os: Linux
   mac:
     properties:
-      os: Mac-10.15
+      os: Mac-12
+      cpu: x86
```

Comment:
Will this break M1s device doctor?

Reply:
No. The mac-12 builds the tool as mac-amd64, same as what mac-10 does. Our m1 bots have been using amd64 version due to the same issue as the swarming python version.

But in the long run, we need to build the arm64 version for M1 bots. Created https://github.com/flutter/flutter/issues/103749 to track.